lib/ecto/query/planner.ex

```diff
@@ -1631,11 +1631,11 @@ defmodule Ecto.Query.Planner do
           {ix, {:parent_as, [], [{:&, meta, [ix]}]}, query}
         end
 
-      %{} ->
-        error!(query, "could not find named binding `parent_as(#{inspect(as)})`")
+      %{} = parent ->
+        get_ix!({:parent_as, meta, [as]}, kind, parent)
```

Comment:
I am wondering if it is better to make it recursive on the way down. I.e. as we go into the subquery, we merge the current as with the parent as.

Reply:
What advantage does this offer over this approach? 

I thought about it, but it would add a cost in all multi-level queries with aliases, while this approach adds a cost only when there is no `parent_as` available at a level. It would indeed give better errors, tho.

Reply:
I think the best is to write them as unit tests in ecto_sql. :)